scripts/experiments_images/SkipGANomaly.jl

```diff
@@ -0,0 +1,127 @@
+using DrWatson
+@quickactivate
+using ArgParse
+using GenerativeAD
+using GenerativeAD.Models: anomaly_score, generalized_anomaly_score
+using BSON
+using StatsBase: fit!, predict, sample
+
+using Flux
+using MLDataPattern
+
+s = ArgParseSettings()
+@add_arg_table! s begin
+   "max_seed"
+        required = true
+        arg_type = Int
+        help = "seed"
+    "dataset"
+        required = true
+        arg_type = String
+        help = "dataset"
+    "anomaly_classes"
+        arg_type = Int
+        default = 10
+        help = "number of anomaly classes"
+end
+parsed_args = parse_args(ARGS, s)
+@unpack dataset, max_seed, anomaly_classes = parsed_args
+
+modelname = "Conv-SkipGANomaly"
+
+function sample_params()
+    argnames = (:num_filters, :extra_layers, :lr, :batch_size,
+                :iters, :check_every, :patience, :lambda,)
+    options = (
+               [2^x for x=2:6],
+               [0:3 ...],
+               [0.0001:0.0001:0.001..., 0.002:0.001:0.01...],
+               [2^x for x=2:8],
+               [10000],
+               [30],
+               [10],
+               [0.9],
+               )
+    w = (weights= sample([1,10:10:90...],3),)
+    return merge(NamedTuple{argnames}(map(x->sample(x,1)[1], options)), w)
+end
+
+"""
+	function fit(data, parameters)
+
+parameters => type named tuple with keys
+	num_filters   - number of kernels/masks in convolutional layers
+	extra_layers  - number of additional conv layers in discriminator
+	lr            - learning rate for optimiser
+	iters         - number of optimisation steps (iterations) during training
+	batch_size    - batch/minibatch size
+
+Note:
+	data = load_data("MNIST")
+	(x_train, y_train), (x_val, y_val), (x_test, y_test) = data
+"""
+function fit(data, parameters)
+	# define models (Generator, Discriminator)
+	model, _ = GenerativeAD.Models.SkipGANomaly_constructor(parameters)
+
+	try
+		global info, fit_t, _, _, _ = @timed fit!(model |>gpu , data, parameters)
+	catch e
+		println("Error caught.")
+		return (fit_t = NaN,), []
+	end
+
+	training_info = (
+		fit_t = fit_t,
+		model = (model |> cpu),
```

Comment:
Save model only once. Probably in the fit function in case of a successful fit. That means that the fit function also needs to know the savepath. The filename should consist of the parameters used in the construction of the model - it should not contain the parameters of the anomaly scoring functions. Test whether this design does not cause trouble somewhere else, e.g. in the check params function.